Generate.py

```diff
@@ -321,13 +321,29 @@ def roll_percentage(percentage: Union[int, float]) -> bool:
     return random.random() < (float(percentage) / 100)
 
 
-def update_weights(weights: dict, new_weights: dict, type: str, name: str) -> dict:
+def update_weights(weights: dict, new_weights: dict, update_type: str, name: str) -> dict:
     logging.debug(f'Applying {new_weights}')
-    new_options = set(new_weights) - set(weights)
-    weights.update(new_weights)
+    cleaned_weights = {}
+    for option in new_weights:
+        if option.startswith("+"):
+            option_name = option[1:]
+            new_value = new_weights[option]
+            if option_name in weights:
+                if isinstance(new_value, (set, dict)):
+                    new_value.update(weights[option_name])
```

Comment:
I think this is wrong. The new should overwrite the old, but the code here has the old overwrite the new. I also swapped them in the suggestion below.